src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -107,23 +107,30 @@ internal void BindParameters(Collection<CommandParameterInternal> parameters)
                         //    windbg  -k com:port=\\devbox\pipe\debug,pipe,resets=0,reconnect
                         // The parser produced an array of strings but marked the parameter so we
                         // can properly reconstruct the correct command line.
-                        bool usedQuotes = false;
+                        StringConstantType stringConstantType = StringConstantType.BareWord;
                         ArrayLiteralAst arrayLiteralAst = null;
                         switch (parameter?.ArgumentAst)
                         {
                             case StringConstantExpressionAst sce:
```

Comment:
I didn't realise we already supported this kind of layering violation in the runtime. Given that this is already the case, I suppose this feature might make more sense than I initially imagined.